pkg/controller/clone/planner.go

```diff
@@ -0,0 +1,522 @@
+		if err := p.watchSnapshots(ctx, args.Log); err != nil {
+			return nil, err
+		}
+	}
+
+	if IsDataSourcePVC(args.DataSource.Spec.Source.Kind) {
+		if args.Strategy == cdiv1.CloneStrategyHostAssisted {
+			args.Log.V(3).Info("Planning host assisted clone from PVC")
+
+			return p.planHostAssistedFromPVC(ctx, args)
+		} else if args.Strategy == cdiv1.CloneStrategySnapshot {
+			args.Log.V(3).Info("Planning snapshot clone from PVC")
+
+			return p.planSnapshotFromPVC(ctx, args)
+		} else if args.Strategy == cdiv1.CloneStrategyCsiClone {
+			args.Log.V(3).Info("Planning csi clone from PVC")
+
+			return p.planCSIClone(ctx, args)
+		}
+	}
+
+	return nil, fmt.Errorf("unknown strategy/source %s", string(args.Strategy))
+}
+
+// Cleanup cleans up after a clone op
+func (p *Planner) Cleanup(ctx context.Context, log logr.Logger, owner client.Object) error {
+	log.V(3).Info("Cleaning up for obj", "obj", owner)
+
+	for _, lt := range listTypesToDelete {
+		ls, err := labels.Parse(fmt.Sprintf("%s=%s", p.OwnershipLabel, string(owner.GetUID())))
+		if err != nil {
+			return err
+		}
+
+		lo := &client.ListOptions{
+			LabelSelector: ls,
+		}
+		if err := cc.BulkDeleteResources(ctx, p.Client, lt, lo); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *Planner) watchSnapshots(ctx context.Context, log logr.Logger) error {
+	p.watchMutex.Lock()
+	defer p.watchMutex.Unlock()
+	if p.watchingSnapshots {
+		return nil
+	}
+
+	vsl := &snapshotv1.VolumeSnapshotList{}
+	lo := &client.ListOptions{Limit: 1}
+	if err := p.Client.List(ctx, vsl, lo); err != nil {
+		if meta.IsNoMatchError(err) {
+			return nil
+		}
+	}
+
+	if err := p.watchOwned(log, &snapshotv1.VolumeSnapshot{}); err != nil {
+		return err
+	}
+
+	log.V(3).Info("watching volumesnapshots now")
+	p.watchingSnapshots = true
+
+	return nil
+}
+
+func (p *Planner) watchOwned(log logr.Logger, obj client.Object) error {
+	objList := p.RootObjectType.DeepCopyObject().(client.ObjectList)
+	if err := p.Controller.Watch(&source.Kind{Type: obj}, handler.EnqueueRequestsFromMapFunc(
+		func(obj client.Object) (reqs []reconcile.Request) {
+			uid, ok := obj.GetLabels()[p.OwnershipLabel]
+			if !ok {
+				return
+			}
+			matchingFields := client.MatchingFields{
+				p.UIDField: uid,
+			}
+			if err := p.Client.List(context.Background(), objList, matchingFields); err != nil {
+				log.Error(err, "Unable to list resource", "matchingFields", matchingFields)
+				return
+			}
+			sv := reflect.ValueOf(objList).Elem()
+			iv := sv.FieldByName("Items")
+			for i := 0; i < iv.Len(); i++ {
+				o := iv.Index(i).Addr().Interface().(client.Object)
+				reqs = append(reqs, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Namespace: o.GetNamespace(),
+						Name:      o.GetName(),
+					},
+				})
+			}
+			return
+		}),
+	); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *Planner) strategyForSourcePVC(ctx context.Context, args *ChooseStrategyArgs) (*cdiv1.CDICloneStrategy, error) {
```

Comment:
Probably just preference but I miss the verb in several functions. For example, I think `getStrategyForSourcePVC` sounds better and it's more informative than just `strategyForSourcePVC`.

Reply:
perhaps this is a bit pedantic, but I like the prefix get/set for simple operations.  simply returning a value that is already set/stored somewhere for example.  This is actually computing a value that is not stored anywhere.  I will add 'compute' prefix